pages/publicaciones/pagina/[page_number].tsx

```diff
@@ -0,0 +1,81 @@
+import type { GetStaticPaths, GetStaticProps } from 'next';
+import type { DualPost } from '../../../lib/types';
+import PostPreview from '../../../components/PostPreview';
+import { getAllPosts } from '../../../lib/getAllPosts';
+import PageWrapper from '../../../components/PageWrapper';
+import Paginator from '../../../components/Paginator';
+import { paginate } from '../../../lib/helpers';
+
+export const getStaticProps: GetStaticProps = async (context) => {
+  const allPosts = await getAllPosts();
+  const whichPage = Number(context.params?.page_number);
+  const thisPagePosts = paginate(
+    allPosts.sort(
+      (a, b) => new Date(b.createdAt).getTime() - new Date(a.createdAt).getTime(),
+    ),
+    whichPage,
+    8,
+  );
+
+  return {
+    props: {
+      posts: thisPagePosts,
+      pageCount: Math.ceil(allPosts.length / 8),
+      pageNum: whichPage,
+    },
+  };
+};
+
+export const getStaticPaths: GetStaticPaths = async () => {
```

Comment:
long, long term, i would like to explore some options for reducing the language-induced duplication in these (and other) next page templates.  i'm not sure there is an elegant way to do that, but there might be, and i feel like these files that are almost exactly the same are ripe for us to make some dumb mistake down the road where only one of them gets updated.  

for another time though...

Reply:
Yeah I've thought about that several times, it would definitely be really nice if there was a good way to do it.